src/arrayMethodJoin.js

```diff
@@ -4,8 +4,30 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let correctSeparator = separator;
+
+    if (separator === null) {
+      correctSeparator = 'null';
+    }
```

Comment:
when you concatenate `string` with `null` or `undefined` you will get `string` so no need to do this additional manipulations
```suggestion
```